pallets/salp/src/lib.rs

```diff
@@ -971,6 +981,62 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// transfer asset to statemine
+		#[pallet::weight(T::WeightInfo::contribute())]
+		pub fn transfer_statemine_assets(
+			origin: OriginFor<T>,
+			amount: BalanceOf<T>,
+			asset_id: u32,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let origin_location = T::AccountIdToMultiLocation::convert(who.clone());
+			let mut assets = MultiAssets::new(); // VersionedMultiAssets::V1(MultiAssets::new(MultiAsset))
+			let amount_128 =
+				TryInto::<u128>::try_into(amount).map_err(|_| Error::<T>::ConvertFailure)?;
+			let statemine_asset = MultiAsset {
+				id: AssetId::Concrete(MultiLocation::new(
+					1,
+					Junctions::X2(
+						Junction::Parachain(1000),
```

Comment:
Based on https://github.com/paritytech/cumulus/pull/806
Why is this id necessary? Within Statemine the asset is uniquely identified by its index.

Reply:
@apopiak  some refact as you mentioned
https://github.com/bifrost-finance/bifrost/pull/420/commits/97476948afda895313ad399048d0ba9471f3fcd9

The problem is that in `InitiateReserveWithdraw` there is some [reanchor asset  logic](https://github.com/paritytech/polkadot/blob/24758c5ab5e6c7f99e9af08e583f6084f5d087a6/xcm/xcm-executor/src/lib.rs#L410) which will add parent and our parachain id as prefix in assets. so when it arrives to dest statemine will still become `(parents:1,X2(parachain:para_id,GeneralIndex:asset_id))` and statemine should understand and handle it if we would like to support such kind of transfer

Reply:
Ok, understood.

Reply:
I'm sorry, I think I misled you here. I was thinking of the `WithdrawAsset` that will be sent to Statemine and not the `InitiateReserveWithdraw`. Your initial code should be correct and be reanchored correctly.
Can you elaborate again why it does not work?

Reply:
@apopiak what I do in [9747694](https://github.com/bifrost-finance/bifrost/commit/97476948afda895313ad399048d0ba9471f3fcd9) is just prove to you it does not work to use no matter `./GeneralIndex(42)` or `../Parachain(1000)/GeneralIndex(42)` in `WithdrawAsset`, the reason as I mentioned before after [asset reanchored logic] in `InitiateReserveWithdraw` assets will be `../Parachain(1000)/GeneralIndex(42)` when it arrives to statemine